atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionManager.java

```diff
@@ -373,7 +373,7 @@ protected final CommitTimestampLoader createCommitTimestampLoader(
                 knowledge);
     }
 
-    protected final KeyValueSnapshotReader createDefaultSnapshotReader(
+    protected KeyValueSnapshotReader createDefaultSnapshotReader(
```

Comment:
this is the only method I need to override, so I can provide a custom SnapshotReader implementation. We can service load a SnapshotReaderFactory instead of the TransactionManagerFactory.